docs/sources/source/tutorials/old_callbacks.rst

```diff
@@ -0,0 +1,206 @@
+
+    from functools import partial
+
+    loggercallback = nemo.core.CheckpointCallback(
+        # The folder to save checkpoints
+        # Note: Neural Factory automatically creates a checkpoint folder
+        folder=neural_factory.checkpoint_dir,
+        # If None, CheckpointCallback will attempt to load from folder
+        # at the beginning of training.
+        # Else, CheckpointCallback will attempt to load from load_from_folder
+        load_from_folder=None,
+        # Checkpointing frequency in steps
+        step_freq=-1,
+        # Checkpointing frequency in epochs
+        epoch_freq=-1,
+        # Number of checkpoints to keep
+        checkpoints_to_keep=4,
+        # If True, CheckpointCallback will raise an Error if restoring fails
+        force_load=False
+    )
+
+EvaluatorCallback
+-----------------
+EvaluatorCallback is used during evaluation to log evaluation
+metrics to screen and tensorboard. EvaluatorCallback requires three arguments:
+eval_tensors, user_iter_callback, user_epochs_done_callback. Similar to
+SimpleLossLoggerCallback, eval_tensors is a list of NMTensors whose values
+we want to obtain during evaluation.
+
+user_iter_callback is a function that is called after each batch during
+evaluation. It is always passed two arguments: values_dict, and global_var_dict.
+values_dict is a dictionary containing NMTensor names as keys, and the evaluated
+tensor as values for that batch. It's main job is to copy the relevant evaluated
+tensors from values_dict to global_var_dict as global_var_dict is saved
+between batches and passed to the final user_epochs_done_callback function.
+
+user_epochs_done_callback is a function that accepts global_var_dict. It's job
+is to log relevant information to the screen such as the evaluation loss.
+
+For simple logging of scalar values to tensorboard, user_epochs_done_callback
+should return a dictionary with strings as keys and scalar tensors as values.
+This tag -> value dictionary will be parsed and each element will be logged
+to tensorboard if a tensorboard writter object is declared.
+
+To enable more complex tensorboard logging such as images or audio,
+EvaluatorCallback must be passed tb_writer_func at initialization. This
+function must accept a
+`tensorboardX.SummaryWriter <https://tensorboardx.readthedocs.io/en/latest/tensorboard.html>`_
+, whatever is returned from user_epochs_done_callback, and the current step.
+We recommend for user_epochs_done_callback to simply return the global_var_dict
+for tb_writer_func to consume. The user must log all data of interest inside
+tb_writer_func including scalars that would otherwise be logged if
+tb_writer_func was not passed to EvaluatorCallback.
+
+You can also log your evaluation metrics into Weights & Biases experiment trackers.
+To do so, please setup these parameters. Also make sure wandb is installed and you did ``wandb login``.
+
+- wandb_name: W&B experiment name
+- wandb_project: W&B project name
```

Comment:
Shouldn't that go into the next section about Wandb?

Reply:
Are these changes (1/3) important? Note that this is old_callbacks.rst, which covers the deprecated callbacks and are planned for removal in v0.12.